canonical/grafana-agent-operator
# feat(alerting): Add up alert rule with anti-flap and absence detection

resolves #143 

## Issue

Grafana Agent does not currently provide an alert that detects the host being down.


## Solution

Add a check, and make it also detect absence of metrics.


## Context



## Testing Instructions

1. Deploy COS
2. Deploy a machine (`juju deploy ubuntu`)
3. Deploy Grafana Agent and relate it to `ubuntu`.
4. Relate the agent to COS
5. Verify that the alert rule has made it over
6. Turn off the host
7. Wait 5 minutes
8. Verify that the alert is firing


## Upgrade Notes
